test/typescript-tests/testTypes.ts

```diff
@@ -2543,6 +2543,12 @@ Statistics functions' return types
   >()
 
   expectTypeOf(math.quantileSeq([1, 2, 3], 0.75)).toMatchTypeOf<number>()
+  expectTypeOf(math.quantileSeq([1, 2, 3, 4, 5], [0.25, 0.75])).toMatchTypeOf<
+    MathArray | MathScalarType
+  >()
+  expectTypeOf(
+    math.quantileSeq([1, 2, 3, 4, 5], [0.25, 0.75]) as number[]
```

Comment:
FYI this doesn't test anything because you check for a type that you previously asserted. :)
I may be available to prepare a PR for the quantileSeq function this weekend if you are interested!

Reply:
@tafaust This tests that it can be cast without throwing an error. See discussion in https://github.com/josdejong/mathjs/issues/3197#issuecomment-2182510802

See L2546 for the equivalent test checking the output type matches the more generic correct type.

Reply:
Yeah, so what this PR does is making 

```ts
math.quantileSeq([1, 2, 3, 4, 5], [0.25, 0.75]) as number[]
``` 

valid TS, rather than requiring:

```ts
math.quantileSeq([1, 2, 3, 4, 5], [0.25, 0.75]) as unknown as number[]
```

It would be nice of course to improve the types further, so that the casting to `as number[]` is not needed at all. A PR would be welcome :)